essentials-module-dynmap/src/main/kotlin/io/github/nyliummc/essentials/entities/FabricDynmapServer.kt

```diff
@@ -0,0 +1,156 @@
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package io.github.nyliummc.essentials.entities
+
+import net.minecraft.block.SignBlock
+import net.minecraft.server.MinecraftServer
+import net.minecraft.text.LiteralText
+import net.minecraft.util.Identifier
+import net.minecraft.util.math.BlockPos
+import net.minecraft.util.registry.Registry
+import net.minecraft.world.dimension.DimensionType
+import org.dynmap.DynmapChunk
+import org.dynmap.DynmapWorld
+import org.dynmap.common.DynmapListenerManager
+import org.dynmap.common.DynmapPlayer
+import org.dynmap.common.DynmapServerInterface
+import org.dynmap.utils.MapChunkCache
+import java.util.HashSet
+import java.util.concurrent.Callable
+import java.util.concurrent.Future
+
+
+class FabricDynmapServer(private val minecraftServer: MinecraftServer) : DynmapServerInterface() {
+    override fun scheduleServerTask(run: Runnable?, delay: Long) {
+        System.out.printf("Task requested with delay %d%n", delay)
+    }
+
+    override fun <T> callSyncMethod(task: Callable<T>?): Future<T>? {
+        println("Sync method requested")
+        return null
+    }
+
+    override fun getOnlinePlayers(): Array<DynmapPlayer?>? {
+        return minecraftServer.playerManager.playerList.map(::FabricDynmapOnlinePlayer).toTypedArray()
+    }
+
+    override fun reload() {
+        TODO("Not yet implemented")
+    }
+
+    override fun getPlayer(name: String?): DynmapPlayer? {
+        return minecraftServer.playerManager.getPlayer(name)?.let(::FabricDynmapOnlinePlayer)
+    }
+
+    override fun getOfflinePlayer(name: String?): DynmapPlayer? {
+        TODO("Not yet implemented")
+    }
+
+    override fun getIPBans(): Set<String?>? {
+        return HashSet(minecraftServer.playerManager.ipBanList.names.asList())
+    }
+
+    override fun getServerName(): String? {
+        return minecraftServer.serverMotd ?: return "Unknown Server"
+    }
+
+    override fun isPlayerBanned(pid: String?): Boolean {
+        // TODO: Verify pid is the username
+        return minecraftServer.playerManager.userBanList.names.contains(pid)
+    }
+
+    override fun stripChatColor(s: String?): String? {
+        TODO("Not yet implemented")
+    }
+
+    override fun requestEventNotification(type: DynmapListenerManager.EventType?): Boolean {
+        TODO("Not yet implemented")
+    }
+
+    override fun sendWebChatEvent(source: String?, name: String?, msg: String?): Boolean {
+        TODO("Not yet implemented")
+    }
+
+    override fun broadcastMessage(msg: String?) {
+        minecraftServer.playerManager.sendToAll(LiteralText(msg))
+    }
+
+    override fun getBiomeIDs(): Array<String?>? {
+        // TODO: Support translation keys if needed
+        return Registry.BIOME.ids.map(Identifier::toString).toTypedArray()
```

Comment:
Registry sync:

A sponge story

Reply:
Any suggestions on how we can solve registry sync related issues? Though I don't think it'll be a big deal since it's a server-side mod.